src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs

```diff
@@ -111,15 +111,9 @@ public async Task Download()
                             continue;
 
                         var hasLogDateTime = LoggingSource.LogInfo.TryGetDate(filePath, out var logDateTime);
-                        if (hasLogDateTime)
-                        {
-                            if (from != null && logDateTime < from)
-                                continue;
+                        if (hasLogDateTime && from.HasValue && logDateTime < from)
```

Comment:
Shouldn't you just add a `break` here:
https://github.com/ravendb/ravendb/pull/17803/files#diff-9748c98933d5efd768da66944b1361616d417b7a35530001bc9bc0896fc64bd4L120